### PR TITLE
feat: no screenshot block patch

### PR DESCRIPTION
**Pull request by Linus789**
related issues: ReVanced/revanced-patches-template#1522, ReVanced/revanced-patches-template#1480
related integrations PR: https://github.com/revanced/revanced-integrations/pull/285

### CODE REVIEW

File: src/main/kotlin/app/revanced/patches/all/screenshot/noblock/patch/NoScreenshotBlockPatch.kt [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+import app.revanced.patcher.patch.PatchResult
+import app.revanced.patcher.patch.PatchResultSuccess
+import app.revanced.patcher.patch.annotations.Patch
+import app.revanced.patcher.util.proxy.mutableTypes.MutableMethod
+import org.jf.dexlib2.Opcode
+import org.jf.dexlib2.iface.instruction.formats.Instruction35c
+import org.jf.dexlib2.iface.reference.MethodReference
+
+@Patch(false)
+@Name("no-screenshot-block")
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/all/screenshot/noblock/patch/NoScreenshotBlockPatch.kt:19]: Please reflect this change to the package and class name as well
```suggestion
@Name("remove-screenshot-restriction")
```

File: src/main/kotlin/app/revanced/patches/all/screenshot/noblock/patch/NoScreenshotBlockPatch.kt [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+@Description("Disables screenshot blocking.")
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/all/screenshot/noblock/patch/NoScreenshotBlockPatch.kt:20]:
```suggestion
@Description("Removes the restriction of making screenshots.")
```

File: src/main/kotlin/app/revanced/patches/all/screenshot/noblock/patch/NoScreenshotBlockPatch.kt [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+@Version("0.0.1")
+class NoScreenshotBlockPatch : BytecodePatch() {
+
+    private companion object {
+        const val integrationClassDefinition = "Lapp/revanced/all/screenshot/noblock/NoScreenshotBlockPatch;"
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/all/screenshot/noblock/patch/NoScreenshotBlockPatch.kt:25]: Make sure to apply the change in referring code as well.
```suggestion
        const val INTEGRATIONS_CLASS_DESCRIPTOR = "Lapp/revanced/all/screenshot/noblock/NoScreenshotBlockPatch;"
```

File: src/main/kotlin/app/revanced/patches/all/screenshot/noblock/patch/NoScreenshotBlockPatch.kt [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+            }
+        }
+
+        companion object {
+            fun fromMethodReference(methodReference: MethodReference): MethodCall? {
+                for (type in values()) {
+                    if (type.definedClassName == methodReference.definingClass && type.methodName == methodReference.name) {
+                        return type
+                    }
+                }
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/all/screenshot/noblock/patch/NoScreenshotBlockPatch.kt:57]: values().[firstOrNull](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/first-or-null.html) can be used here.

**Review by oSumAtrIX — APPROVED** [review]

File: src/main/kotlin/app/revanced/patches/all/screenshot/noblock/patch/NoScreenshotBlockPatch.kt (hunk 1 above)
Comment (Linus789) [reply src/main/kotlin/app/revanced/patches/all/screenshot/noblock/patch/NoScreenshotBlockPatch.kt:19]: Would `app/revanced/patches/all/screenshot/removerestriction/patch/RemoveScreenshotRestrictionPatch.kt` be okay?

Comment (oSumAtrIX) [reply src/main/kotlin/app/revanced/patches/all/screenshot/noblock/patch/NoScreenshotBlockPatch.kt:19]: Looks fine.

File: src/main/kotlin/app/revanced/patches/all/screenshot/noblock/patch/NoScreenshotBlockPatch.kt (hunk 4 above)
Comment (Linus789) [reply src/main/kotlin/app/revanced/patches/all/screenshot/noblock/patch/NoScreenshotBlockPatch.kt]: everything done

File: src/main/kotlin/app/revanced/patches/all/screenshot/removerestriction/patch/RemoveScreenshotRestrictionPatch.kt [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+                }
+            }
+        }
+
+        companion object {
+            fun fromMethodReference(methodReference: MethodReference): MethodCall? {
+                return values().firstOrNull { search ->
+                    search.definedClassName == methodReference.definingClass && search.methodName == methodReference.name
+                }
+            }
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/all/screenshot/removerestriction/patch/RemoveScreenshotRestrictionPatch.kt:56]:
```suggestion
            fun fromMethodReference(methodReference: MethodReference) = values().firstOrNull { search ->
                search.definedClassName == methodReference.definingClass && search.methodName == methodReference.name
            }
```

**Review by oSumAtrIX — CHANGES_REQUESTED** [review]

**Comment by oSumAtrIX** [comment]
Thanks.

**Pull request merged**